src/service/viewer-impl.js

```diff
@@ -802,6 +802,25 @@ export class Viewer {
   }
 
   /**
+   * Get the fragment from the url or the viewer.
+   * Strip leading '#' in the fragment
+   * @return {!Promise<string|undefined>}
+   */
+  getFragment() {
+    if (!this.isEmbedded_) {
```

Comment:
Should we just ask the viewer to set the fragment to us? That'd be a lot faster than waiting for message channel...